pkg/controller/common/manage_children.go

```diff
@@ -42,6 +46,10 @@ func ApplyUpdate(orig, update *unstructured.Unstructured) (*unstructured.Unstruc
 	if err != nil {
 		return nil, err
 	}
+
+	//prevent setting last applied values in the new object
+	nullifyLastAppliedAnnotation(update)
```

Comment:
should it happen only when SSA apply strategy is used, right ?

Reply:
actually no. This line fixes a bug where if you simply return the "observed" as a "expected" child, that you recursively end up with the whole object in "last applied" annotation, filling up ETCD (you don't want to know why I figured that out 😅 )

Reply:
hmm got it, will try to recreate the issue